### PR TITLE
[WIP] Align with code in LightGBM for Categorical features

### PR DESCRIPTION
**Pull request by Ivanidzo4ka**
Potentially fixes https://github.com/dotnet/machinelearning/issues/1625
Honestly have no idea what I'm doing here, so I would appreciate any feedback.

Need tests.

### CODE REVIEW

File: src/Microsoft.ML.LightGbm/WrappedLightGbmBooster.cs [hunk 1]
```diff
@@ -162,10 +162,10 @@ private static double[] GetDefalutValue(double[] threshold, UInt32[] decisionTyp
             return ret;
         }
 
-        private static bool FindInBitset(UInt32[] bits, int start, int end, int pos)
+        private static bool FindInBitset(UInt32[] bits, int start, int size, int pos)
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.LightGbm/WrappedLightGbmBooster.cs]: [https://github.com/Microsoft/LightGBM/blob/90127b526be8b13bdf840effacde9e61fa765a6f/include/LightGBM/utils/common.h#L827](https://github.com/Microsoft/LightGBM/blob/90127b526be8b13bdf840effacde9e61fa765a6f/include/LightGBM/utils/common.h#L827)
this is original function in LightGBM.

File: src/Microsoft.ML.LightGbm/WrappedLightGbmBooster.cs [hunk 2]
```diff
@@ -174,13 +174,13 @@ private static bool FindInBitset(UInt32[] bits, int start, int end, int pos)
         private static int[] GetCatThresholds(UInt32[] catThreshold, int lowerBound, int upperBound)
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.LightGbm/WrappedLightGbmBooster.cs]: >GetCatThresholds [](start = 29, length = 16)

[https://github.com/Microsoft/LightGBM/blob/babe8b313aad747899a3199fba7bc3eae10e582a/src/io/tree.cpp#L270-L279](https://github.com/Microsoft/LightGBM/blob/babe8b313aad747899a3199fba7bc3eae10e582a/src/io/tree.cpp#L270-L279)
This is closest I could find in LightGBM to our logic.

**Comment by Ivanidzo4ka** [comment]
@codemzs @TomFinley if one of you have spare time...

**Comment by codemzs** [comment]
@Ivanidzo4ka CI legs are failing. Please feel free to fix build failures and then we can review. I'm closing this PR for now.

**Pull request closed without merging**